Fix references for legacy UserGroups in Comments

andreslucena opened the pull request:
#### :tophat: What? Why?

Legacy user groups mentions in comments weren't migrated accordingly after #13969. This PR fixes it. 

#### :pushpin: Related Issues
 
- Fixes #15450

#### Testing

1. Create a v0.30 app and get the UserGroup nickname (and email just in case to check it out later)
```
git checkout release/0.30-stable
git pull origin release/0.30-stable
bundle exec rake development_app
redis-cli flushall
bin/rails runner "ug = Decidim::UserGroup.first ; puts ug.nickname ; puts ug.email"
```
2. Then spin up the app and do a mention with the nickname from the output of this command - **See first screenshot**. On my case the output was:
> Running via Spring preloader in process 947054
> eddy_zboncak
> dwayne@schinner.test
4. Update to v0.31 
```
bin/rails runner "print Rails.application.secret_key_base" > development_app/tmp/local_secret.txt

git checkout develop
cd development_app

# secrets API deprecations
rm config/initializers/decidim.rb config/environments/production.rb
cp ../decidim-generators/lib/decidim/generators/app_templates/storage.yml config/

# # changing Rails defaults
sed -i "s/6\.1/7.2/g" config/application.rb

bin/rails decidim:upgrade
bin/rails db:migrate

# # to actually kick the user group change
bin/rails decidim:upgrade:user_groups:remove
``` 
5. Go to the comment, check that you have the same kind of string `gid://etc/x/y` (on my case `gid://decidim-development-app/Decidim::UserGroup/5 `)   - **See second screenshot**
6. Checkout this branch
7. Run the command for data migration
```
bin/rails data:migrate
```
8. See that the comment is fixed  - **See third screenshot**

### :camera: Screenshots
<img width="1035" height="481" alt="image" src="https://github.com/user-attachments/assets/67f7c736-f6b8-40ed-908a-d4aec494c6fa" />

<img width="1200" height="415" alt="image" src="https://github.com/user-attachments/assets/a7c59235-8f49-43f1-8500-691a7855e199" />

<img width="1027" height="375" alt="image" src="https://github.com/user-attachments/assets/ccf3132e-64f5-4022-9c7e-e72cf4960d97" />



:hearts: Thank you!

andreslucena committed:
Rename rake task spec for decidim-comments

andreslucena committed:
Add task for changing reference for legacy UserGroups in Comments

andreslucena committed:
Call task during the user group removal task

andreslucena committed:
Fix rubocop offenses (automatic)

andreslucena committed:
Fix rubocop offenses (manual)

alecslupu commented on decidim-comments/lib/tasks/upgrade/decidim_comments.rake:
```diff
@@ -24,5 +24,54 @@ namespace :decidim_comments do
+        updated_body = {}
+
+        # Process each locale in the body hash
+        comment.body.each do |locale, text|
```
I think here we have to take into account that when machine translation is involved, you'd need to patch `decidim.body["machine_translations"]` separately.

andreslucena replied:
Fixed on 8e631a1861

andreslucena committed:
Refactor examples to make them clearer and easy to work with the Hashes

andreslucena committed:
Fix bug when machine translations enabled (detected on code review)

andreslucena commented:
@alecslupu this is ready for another round

alecslupu reviewed (commented):
**The fix works as intended for comments**. I have 2 questions ( sorry if they are dumb):
- Why we didn't do this as a data migration (considering we're still fresh with data-migration, i could let this pass, since is bound to the upgrade groups)?
- Do we have this issue only on comments, or there are any other resources having this?

andreslucena committed:
Merge branch 'develop' into fix/user_groups_references

andreslucena committed:
Migrate to data-migrate

andreslucena committed:
Remove call to migrated task

andreslucena committed:
Fix rubocop offenses

andreslucena commented:
> Why we didn't do this as a data migration (considering we're still fresh with data-migration, i could let this pass, since is bound to the upgrade groups)?

I migrated the code just to see the differences and it was really fast.

Also I think we had some rules that relaxed the rubocop configuration for the tasks files. As I had new offenses I fixed them by refactoring. 

I tried it again the whole process just to be sure and it worked well. I updated the Testing instructions in the PR description to better reflect the new flow. 

> Do we have this issue only on comments, or there are any other resources having this?

As far as I know, mentions to UserGroups only work in comments. I double checked with @carolromero last week too.

alecslupu reviewed (approved):
:+1: 

Merging with codecov pipeline failure.

alecslupu commented:
Adding `no-backport` as this PR targeted 0.31 only.
